File: Replace Elements with Greatest Element on Right Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        N = len(arr)
        max_from_right = arr[N-1]
        arr[N-1] = -1
        
        for i in range(N-2,-1,-1):
            temp = arr[i]
            arr[i] = max_from_right
            
            if max_from_right< temp: 
                max_from_right=temp
        
        return arr        

File: Replace Elements with Greatest Element on Right Side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int max = 0;
        for (int i = 0; i < arr.length; i++) {
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] > max) {
                    max = arr[j];
                }
            }
            arr[i] = max;
            max = 0;
        }
        arr[arr.length - 1] = -1;
        return arr;
    }
}
